### PR TITLE
tests: drivers: gpio: gpio_basic_api: Add nrf54h20dk overlays

### PR DESCRIPTION
**Pull request by nordic-krch**
Add overlays for nrf54h20dk_nrf54h20 cpuapp and cpurad.

### CODE REVIEW

**Review by nika-nordic — DISMISSED** [review]

File: modules/hal_nordic/nrfx/nrfx_glue.h [hunk 1]
```diff
@@ -334,8 +334,56 @@ void nrfx_busy_wait(uint32_t usec_to_wait);
 				   NRFX_PPI_GROUPS_USED_BY_802154_DRV | \
 				   NRFX_PPI_GROUPS_USED_BY_MPSL)
 
-/** @brief Bitmask that defines GPIOTE channels that are reserved for use outside of the nrfx library. */
-#define NRFX_GPIOTE_CHANNELS_USED NRFX_GPIOTE_CHANNELS_USED_BY_BT_CTLR
```
Comment (anangl) [inline modules/hal_nordic/nrfx/nrfx_glue.h:338]: We should get rid of `NRFX_GPIOTE_CHANNELS_USED_BY_BT_CTLR` completely and instead allocate GPIOTE channels properly in the Bluetooth controller. Then, all the stuff that you are adding below will be unnecessary.

**Review by anangl — CHANGES_REQUESTED** [review]

File: modules/hal_nordic/nrfx/nrfx_glue.h (hunk 1 above)
Comment (anangl) [reply modules/hal_nordic/nrfx/nrfx_glue.h:338]: Here's my attempt to solve the problem: #73145.

Comment (nordic-krch) [reply modules/hal_nordic/nrfx/nrfx_glue.h]: removed BT_CTRL part since #73145 got merged.

File: modules/hal_nordic/nrfx/nrfx_glue.h [hunk 2]
```diff
@@ -334,6 +334,24 @@ void nrfx_busy_wait(uint32_t usec_to_wait);
...
+#if defined(CONFIG_SOC_SERIES_NRF54HX)
+#define GPIOTE_CH_OWNED_CH(node_id, prop, idx) | BIT(DT_PROP_BY_IDX(node_id, prop, idx))
+
+/* Macro for getting mask of channels assigned to the given core. It shall be used only for GPIOTE
+ * instances which has split ownership.
+ */
+#define GPIOTE_OWNED_CHANNELS(idx)                                                                \
+	(COND_CODE_1(DT_NODE_HAS_PROP(DT_NODELABEL(gpiote##idx), owned_channels),                 \
+	((0 DT_FOREACH_PROP_ELEM(DT_NODELABEL(gpiote##idx), owned_channels, GPIOTE_CH_OWNED_CH))),\
+	(0)))
```
Comment (anangl) [inline modules/hal_nordic/nrfx/nrfx_glue.h:346]: You can use an already existing macro:
https://github.com/zephyrproject-rtos/zephyr/blob/6a070ee165fe39b2fcbfd365102165ab362dce2f/modules/hal_nordic/nrfx/nrfx_config.h#L874-L878
```
#define NRFX_GPIOTE130_CHANNELS_USED ~NRFX_CONFIG_MASK_DT(DT_NODELABEL(gpiote130), owned_channels)

File: modules/hal_nordic/nrfx/nrfx_glue.h [hunk 3]
```diff
@@ -334,6 +334,24 @@ void nrfx_busy_wait(uint32_t usec_to_wait);
...
+
+#if DT_NODE_EXISTS(DT_NODELABEL(gpiote130))
+/** @brief Bitmask that defines GPIOTE130 channels reserved for use outside of the nrfx library. */
+#define NRFX_GPIOTE130_CHANNELS_USED ~GPIOTE_OWNED_CHANNELS(130)
```
Comment (anangl) [inline modules/hal_nordic/nrfx/nrfx_glue.h:350]: `DT_NODE_EXISTS()` is actually redundant, as `DT_NODE_HAS_PROP()` correctly handles non-existing nodes.

**Review by anangl — APPROVED** [review]

**Review by nika-nordic — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**